File: backend/cs/SocialNetApp/Bff/API/Dtos/MessageDto.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Bff.API.Dtos
{
    public readonly record struct MessageDto
    {
        [Required]
        [JsonPropertyName("id")]
        public uint Id { get; init; }

        [Required]
        [JsonPropertyName("authorId")]
        public uint AuthorId { get; init; }

        [JsonPropertyName("message")]
        public string Message { get; init; }

        [JsonPropertyName("created")]
        public DateTime? Created { get; init; }
    }
}

File: backend/go/socialnetapp/bff/internal/handler/dto/messagedto.go
package dto

import "time"

// MessageDto represents dto for a dialog message
type MessageDto struct {
	ID       uint       `json:"id"`
	AuthorId uint       `json:"authorId"`
	UserId   uint       `json:"userId"`
	Message  string     `json:"message"`
	Created  *time.Time `json:"created"`
}
